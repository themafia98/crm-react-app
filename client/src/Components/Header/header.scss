header {
    width: 100%;
    z-index: 1000;
    display: flex;
    background: whitesmoke;
    
    .menu{
        display: flex;
        width: 100%;
        padding: 0;
        justify-content: flex-end;
        flex-wrap: wrap;
        list-style-type: none;
        a {
            
            text-decoration: none;
            font-weight: bold;
            color: rgb(0, 0, 0);
            padding: 10px;
        }

        .selected {
            color :red;
        }
    }
}

.header-flex {
    margin: 0 auto;
    max-width: 90%;
    width: 100%;
    display: flex;
    align-items: center;
};

header.fixed{

    max-width: 100%;
    position: fixed;
    z-index: 100;
    width: 100%;
    top: 0;
    background: whitesmoke;
    display: flex;

}

    
