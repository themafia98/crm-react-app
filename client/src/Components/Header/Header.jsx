import React from 'react';
import { NavLink, withRouter  } from 'react-router-dom';

import Logo from '../Logo/Logo';

import './header.scss';

const Header = ({go, location, history}) => {

    const redirectHome = () => {
        history.push('/');
    };

    return (
        <header>
            <Logo cbRedirectHome = {go && redirectHome} go = {go} path = '/img/logo.png' />
            <ul className = 'menu'>
                <NavLink  to = {'/'}>
                    <li>Главная</li>
                </NavLink >
                <NavLink  to = '/Service'>
                    <li>Услуги</li>
                </NavLink >
                <NavLink  to = '/About'>
                    <li>О нас</li>
                </NavLink >
                <NavLink  to = '/Contact'>
                    <li>Контакты</li>
                </NavLink >
            </ul>
        </header>
    )
}

export default withRouter(Header);