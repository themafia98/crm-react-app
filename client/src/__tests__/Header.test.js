import React from 'react';
import {MemoryRouter} from 'react-router-dom';
import renderer from 'react-test-renderer';
import Header from '../Components/Header/Header';

const HeaderTest = renderer.create(
    <MemoryRouter>
        <Header />
    </MemoryRouter>
)


test('Header test', () => {

    let treeSnapshot = HeaderTest.toJSON();
    expect(treeSnapshot).toMatchSnapshot();
});