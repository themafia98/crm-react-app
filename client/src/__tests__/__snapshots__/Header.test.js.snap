// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header test 1`] = `
<header
  className={null}
>
  <div
    className="container header-flex"
  >
    <div
      className="Icon logo"
    >
      <img
        alt="logo"
        className={null}
        src="/img/logo.png"
      />
    </div>
    <ul
      className="menu"
    >
      <a
        aria-current="page"
        className="selected"
        href="/"
        onClick={[Function]}
        style={Object {}}
      >
        <li>
          Главная
        </li>
      </a>
      <a
        aria-current={null}
        href="/Services"
        onClick={[Function]}
      >
        <li>
          Услуги
        </li>
      </a>
      <a
        aria-current={null}
        href="/About"
        onClick={[Function]}
      >
        <li>
          О нас
        </li>
      </a>
      <a
        aria-current={null}
        href="/Contact"
        onClick={[Function]}
      >
        <li>
          Контакты
        </li>
      </a>
    </ul>
  </div>
</header>
`;
